gcp/website/frontend3/src/search.js

```diff
@@ -0,0 +1,130 @@
+export class ExpandableSearch {
+  constructor() {
+    this.containers = [];
+    
+    this.cleanupExistingInstances();
+    this.setupGlobalListeners();
+    this.initializeSearch();
+  }
+  
+  cleanupExistingInstances() {
+    // If there's a global instance, clean it up
+    if (window.OSVSearchInstance) {
+      if (window.OSVSearchInstance.documentClickHandler) {
+        document.removeEventListener('click', window.OSVSearchInstance.documentClickHandler);
+      }
+      if (window.OSVSearchInstance.documentKeydownHandler) {
+        document.removeEventListener('keydown', window.OSVSearchInstance.documentKeydownHandler);
+      }
+    }
+    
+    // Store this instance globally
+    window.OSVSearchInstance = this;
+  }
+  
+  setupGlobalListeners() {
+    this.documentClickHandler = this.handleDocumentClick.bind(this);
+    this.documentKeydownHandler = this.handleDocumentKeydown.bind(this);
+    
+    document.addEventListener('click', this.documentClickHandler);
+    document.addEventListener('keydown', this.documentKeydownHandler);
+  }
+  
+  handleDocumentClick(e) {
+    this.containers.forEach(container => {
+      if (container && container.element && 
+          !container.element.contains(e.target) && 
+          container.form.classList.contains('active')) {
+        this.closeSearch(container);
+      }
+    });
+  }
+  
+  handleDocumentKeydown(e) {
+    // Close any open search on escape key
+    if (e.key === 'Escape') {
+      this.containers.forEach(container => {
+        if (container && container.form && 
+            container.form.classList.contains('active')) {
+          this.closeSearch(container);
+        }
+      });
+    }
+  }
+
+  initializeSearch() {
+    this.initializeSearchContainer('.search-container-nav');
+  }
+  
+  initializeSearchContainer(containerSelector) {
+    const searchContainer = document.querySelector(containerSelector);
+    if (!searchContainer) {
+      return;
+    }
+
+    const searchForm = searchContainer.querySelector('.search-form');
+    const searchToggle = searchContainer.querySelector('.search-toggle');
+    const searchInput = searchContainer.querySelector('.search-input');
+    
+    if (!searchForm || !searchToggle || !searchInput) {
+      return;
+    }
+
+    const containerInfo = {
+      element: searchContainer,
+      form: searchForm,
+      toggle: searchToggle,
+      input: searchInput
+    };
+    
+    this.containers.push(containerInfo);
+
+    const toggleHandler = (e) => {
+      e.preventDefault();
+      const isActive = searchForm.classList.contains('active');
+      
+      if (isActive) {
+        this.closeSearch(containerInfo);
+      } else {
+        this.openSearch(containerInfo);
+      }
+    };
+    
+    searchToggle.removeEventListener('click', toggleHandler);
+    searchToggle.addEventListener('click', toggleHandler);
+    searchToggle.setAttribute('data-search-initialized', 'true');
```

Comment:
```suggestion
    if (!searchToggle.hasAttribute('data-search-initialized')) {
      searchToggle.addEventListener('click', toggleHandler);
      searchToggle.setAttribute('data-search-initialized', 'true');
    }
```
We can use the data attribute here to prevent attaching the same listener multiple times :) - because toggleHandler is defined inside the `initializeSearchContainer` method, every time  `initializeSearchContainer`  is called, a new `toggleHandler` function is created in memory.